packages/core/src/getters/query-params.ts

```diff
@@ -68,12 +68,21 @@ const getQueryParamsTypes = (
 
     if (resolvedValue.isEnum && !resolvedValue.isRef) {
       const enumName = queryName;
-
-      const enumValue = getEnum(
-        resolvedValue.value,
-        enumName,
-        resolvedValue.originalSchema?.['x-enumNames'],
-      );
+      let enumValue: string;
+
+      if (context.override.useNativeEnums) {
```

Comment:
This looks pretty good but I am wondering if the getEnum method can just be passed this config so that getEnum returns either current or the native enum?  It would make all these if else's cleaned up?

Reply:
You are absolutely right, the ifs look terribly ugly. I've performed the changes requested.